### PR TITLE
Fix LIF oversampling.

### PR DESCRIPTION
**Pull request by thorstenhater**
Fixes an issue where LIF sampling across multiple epochs causes repetition of the same sampling event et ergo
multiple samples for the same time, viz
```
0 v0
0 v0
1 v1
0 v0
1 v1
2 v2
...
```

### CODE REVIEW

File: arbor/lif_cell_group.cpp [hunk 1]
```diff
@@ -140,7 +144,11 @@ void lif_cell_group::advance_cell(time_type tfinal, time_type dt, cell_gid_type
             if (delta == 0) continue;
             n_values += delta;
             // only exact sampling: ignore lax and never look at policy
-            for (auto t: times) samples.emplace_back(t, hdl);
+            for (auto t: times) {
+                if (t > last_time_sampled_[lid]) {
```
Comment (boeschf) [inline arbor/lif_cell_group.cpp:148]: is this check necessary considering you create the scheduled events with the last sampled time before?

File: arbor/lif_cell_group.cpp [hunk 2]
```diff
@@ -123,11 +127,15 @@ void lif_cell_group::advance_cell(time_type tfinal, time_type dt, cell_gid_type
...
         std::lock_guard<std::mutex> guard(sampler_mex_);
         for (auto& [hdl, assoc]: samplers_) {
-            if (assoc.probeset_ids.empty()) continue; // No need to generate events
-            // Construct sampling times
-            const auto& times = util::make_range(assoc.sched.events(t, tfinal));
+             // No need to generate events
+            if (assoc.probeset_ids.empty()) continue;
+            // Construct sampling times, might give us the last time we sampled, so skip that.
+            auto times = util::make_range(assoc.sched.events(tlast, tfinal));
+            if (!times.empty() && times.front() == tlast) times.left++;
```
Comment (boeschf) [inline arbor/lif_cell_group.cpp:137]: can we be sure that `tlast` doesn't appear twice or more times?

Comment (thorstenhater) [reply arbor/lif_cell_group.cpp:137]: Good point.

Comment (thorstenhater) [reply arbor/lif_cell_group.cpp]: ...and fixed.

**Review by boeschf — APPROVED** [review]
looks good, thanks

**Pull request merged**